### ISSUE TITLE
NullPointerException

### ISSUE THREAD

**Issue by Teg-MC** [TechzoneMC/NPCLib#6]
Getting this error when I try to spawn an NPC with a skin http://pastebin.com/RyZyFbd2. Using NPCLib 1.1.1-SNAPSHOT and this(http://i.teg.pw/h6n6o.png) Spigot version. The NPC spawns just fine if I don't try to set a skin.

Getting the same error using this(http://i.teg.pw/ig2e8.png) version of Spigot as well.

**Comment by Techcable** [comment]
Could i please see your code?
This is due to commit 1cddc524ae

**Comment by Teg-MC** [comment]
Code: http://pastebin.com/d7MjxiFX

NPCController line 64 (from the error) is 'npc.spawn(location);'

**Comment by Techcable** [comment]
Where is someValidUUID?
Also could you open a new issue for the wierd looking npcs?

**Comment by Teg-MC** [comment]
someValidUUID is just a field with a UUID in it

**Comment by Teg-MC** [comment]
I did attempt to find the cause of this issue, and I believed it to be that the lib was being rate-limited.

When you call setSkin(UUID uuid) it requests the skin from Mojang, but doesn't set it because the entity isn't spawned yet. When you call spawn(Location location) right after it requests the skin again, but Mojang rate-limits the request, so the PlayerProfile it returns is null, hence the NPE.

I decided to remove the trySetSkin() from the setSkin(UUID uuid) method and move the trySetSkin() in the spawn(Location location) to after the entity is spawned, but for some reason the skin still didn't work.

http://i.teg.pw/y52za.png
